libraries/AP_BLHeli/AP_BLHeli.cpp

```diff
@@ -539,6 +551,9 @@ void AP_BLHeli::msp_process_command(void)
         break;
     }
     default:
+        // generally we can't see debug messages while using passthrough - this is a sneaky way of
+        // telling that we were asked for a command we don't understand
+        AP::internalerror().error(AP_InternalError::error_t::flow_of_control, msp.cmdMSP);
```

Comment:
I don't quite understand your comment here - but we definitely shouldn't be throwing an internal error based on external input we don't understand (it's not an error internal to ArduPilot).

Perhaps I'm misunderstanding why this is here.

Reply:
This code is all to support passthrough. When using passthrough you cannot see anything from the console because the USB connection is being used for passthrough. The current bug came about because the external tool changed the MSP messages it needed to a set that included ones we don't support. No way of knowing as a user - it just hangs. This at least catches the problem if it happens again, because we will get an internal error displayed after disconnecting if this happens again. It *is* an internal error because it means we are missing parts of the protocol definition.

Reply:
> This code is all to support passthrough. When using passthrough you cannot see anything from the console because the USB connection is being used for passthrough. The current bug came about because the external tool changed the MSP messages it needed to a set that included ones we don't support. No way of knowing as a user - it just hangs. This at least catches the problem if it happens again, because we will get an internal error displayed after disconnecting if this happens again. It _is_ an internal error because it means we are missing parts of the protocol definition.

OK, this is the second time this sort of thing has come up recently.  tridge suggested we rename this to "CriticalError" last time it came up :-)

This is the problem.  Internal errors are *supposed* to be for "land right now, do not fly again until you have fixed the bug" sorts of errors.  Not for problems that arise because some external entity happens to have changed a protocol definition and we haven't caught up yet.  Look at the list of InternalErrors and try to tell me with a straight face this is in the same category as those bugs :-)

We need to find a better way of getting across these problems which doesn't dilute what InternalError is supposed to be for.

It would be trivial to add a logged Event (`AP::logger().log_event` or whatever), but I don't think that will get the effect you're after - a persistent warning to the user via mavlink?

Reply:
Yes, I am trying to make it easy for the user.

Note that you can only use passthrough while sitting on the ground. There is no way you can be flying. So it's also safe to do it this way.

Reply:
i agree with @peterbarker on this one

Reply:
Ok, removed